### PR TITLE
Unit tests and minor fixes

### PR DESCRIPTION
**Pull request by ezequielc**
## Description

- Unit tests
- Reverts sorting schedules by name (use last modified instead)
- Shows error message on Schedule change failure
- Keep event name as 'Schedule Updated' for consistency/reporting

## Motivation and Context
Why is this change required? What problem does it solve?

## How Has This Been Tested?
Describe in detail how you tested your changes. Include details of your testing environment and link(s) for reviewers to validate.

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
If any Release Checklist items were intentionally skipped, please provide which ones and the reasons why

### CODE REVIEW

File: web/scripts/displays/controllers/ctr-display-details.js [hunk 1]
```diff
@@ -196,7 +196,10 @@ angular.module('risevision.displays.controllers')
         } else {
           var shouldSkipAddressValidation = !addressService.isAddressFormDirty($scope.displayDetails) && !$scope.displayDetails.useCompanyAddress.$dirty;
           return displayFactory.updateDisplay(shouldSkipAddressValidation).then(function() {
-            scheduleFactory.addToDistribution($scope.display, $scope.selectedSchedule);
+            scheduleFactory.addToDistribution($scope.display, $scope.selectedSchedule).catch(function() {
+              displayFactory.errorMessage = scheduleFactory.errorMessage;
+              displayFactory.apiError = scheduleFactory.apiError;
```
Comment (alex-deaconu) [inline web/scripts/displays/controllers/ctr-display-details.js:201]: @ezequielc I'm not a huge fan of doing this, but I don't have any better suggestions. Thanks.

Comment (alex-deaconu) [reply web/scripts/displays/controllers/ctr-display-details.js:201]: Maybe we can move `addToDistribution` to the `displayFactory`? Just a thought. Thanks.

Comment (ezequielc) [reply web/scripts/displays/controllers/ctr-display-details.js:201]: @alex-deaconu An alternative I actually tried was to handle `scheduleFactory` errors in the display partial as well. But that would generate a lot of additional conditions to check, like cleaning up `scheduleFactory` every time we load displays page or make requests on `displayFactory`. So this was a much simpler solution.

Another idea I thought was to have a generic $scope.errorMessage and watch/handle both factories. But I don't see any harm on doing the way I did, and is a simpler approach. Thanks

Comment (alex-deaconu) [reply web/scripts/displays/controllers/ctr-display-details.js:201]: @ezequielc sounds good, thanks!

**Pull request merged**